src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java

```diff
@@ -141,6 +157,27 @@ public void shouldNotAuthenticateForNonAdminUserWithoutOBOPermission() {
         }
     }
 
+    @Test
+    public void shouldNotIncludeHostMappingInOBOToken() {
+        String oboToken = generateOboToken(OBO_USER_NAME_WITH_HOST_MAPPING, DEFAULT_PASSWORD);
+
+        String[] splitToken = oboToken.split("\\.");
+
+        String unsignedToken = splitToken[0] + "." + splitToken[1] + ".";
+        Claims claims = Jwts.parserBuilder().build().parseClaimsJwt(unsignedToken).getBody();
+        Object er = claims.get("er");
+        EncryptionDecryptionUtil encryptionDecryptionUtil = new EncryptionDecryptionUtil(encryptionKey);
+        String rolesClaim = encryptionDecryptionUtil.decrypt(er.toString());
+        List<String> roles = Arrays.stream(rolesClaim.split(","))
+            .map(String::trim)
+            .filter(s -> !s.isEmpty())
+            .collect(Collectors.toUnmodifiableList());
+
+        Assert.assertTrue(roles.contains("host_mapping_role"));
```

Comment:
I don't think this test is quite capturing the issue.

Imagine the following configuration:

```
internal_users:

craig

roles_mapping:

role1:
  users:
    - craig

role2:
  hosts:
    - 127.0.0.1
```

- If the user `craig` makes a request from 127.0.0.1 to search on index `logs`, the search request will be evaluated against `role1` and `role2`

- If the user `craig` makes a request from 127.0.0.1 to create an obo token, the IP address the request originates from should be ignored and the obo token's roles claim should only include `role1`

Reply:
Got it! I will fix the setup in the next commit~ Thanks for the example.

Reply:
Fixed